cdm/core/src/main/java/ucar/nc2/filter/Standardizer.java

```diff
@@ -0,0 +1,80 @@
+package ucar.nc2.filter;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.math.stat.descriptive.SummaryStatistics;
+import ucar.ma2.Array;
+import ucar.ma2.DataType;
+import ucar.ma2.IndexIterator;
+import ucar.nc2.dataset.VariableDS;
+
+public class Standardizer {
+
+  private final ScaleOffset scaleOffset;
+  private static final String name = "Standardizer";
+  private static final int id = -1;
+  private final double mean;
+  private final double stdDev;
+
+  public static Standardizer createFromVariable(VariableDS var) {
+    try {
+      Array arr = var.read();
+      DataType type = var.getDataType();
+      return new Standardizer(arr, type);
+    } catch (IOException e) {
+      return new Standardizer(0.0, 1.0, var.getDataType());
+    }
+  }
+
+  private Standardizer(double mean, double stdDev, DataType type) {
+    this.mean = mean;
+    this.stdDev = stdDev;
+    Map<String, Object> props = new HashMap<>();
+    props.put("offset", mean);
+    props.put("scale", 1 / stdDev);
+    props.put("dtype", type);
+    scaleOffset = new ScaleOffset(props);
+  }
+
+  public Standardizer(Array arr, DataType type) {
+    this(calculateMean(arr), calculateStandardDeviation(arr), type);
```

Comment:
One optimization we could do (okay with me to wait on this til later though) is to only loop over the only data once and get both the mean and standard deviation at the same time

Reply:
Added a helper function and the constructor of Standardize was refactored, so it should now loop only once through the data for the calculations.